crates/namada/src/ledger/protocol/mod.rs

```diff
@@ -740,16 +765,20 @@ where
                     );
                 }
 
-                let masp_ref = namada_tx::action::get_masp_section_ref(*state)
-                    .map_err(Error::StateError)?;
+                let masp_section_ref =
+                    match namada_tx::action::get_masp_section_ref(*state)
+                        .map_err(Error::StateError)?
+                    {
+                        Some(masp_tx_id) => Either::Left(masp_tx_id),
+                        None => Either::Right(*first_tx.cmt.data_sechash()),
```

Comment:
Should we check if the ibc action has been written in this case to be slightly more safe?

Reply:
fixed to check it: 45d2798